queries/chg_vers_first_bill_run_after.md

```diff
@@ -0,0 +1,142 @@
+# Charge version first bill run after
+
+- **Business**
+- **2023-10-09**
+- [WATER-4136](https://eaflood.atlassian.net/browse/WATER-4136)
+
+> The B&D team wants a 'billed up to' date against a charge version to help them manage billing. It was something they had in NALD but has been lost in WRLS.
+>
+> We think there might be a way to see what was the first bill run created after a charge version is created by comparing their `date_created` fields.
+
+## Query
+
+```sql
+/**
+ * For each charge version get details for the first annual and supplementary bill
+ * run (billing_batch) created after the charge version with the same scheme and region.
+ *
+ * The way to understand the query is to follow how the data is built up starting at
+ * the centre and then working out. Follow the numbered comments [1] to [5].
+ *
+ * It is as complex as it is because
+ *
+ * - we want to identify the bill run with the earliest date created which also matches
+ *   the charge version. Functions like MIN() will just return the smallest value in a
+ *   table
+ * - we wanted to avoid just having first bill run and instead split out first possible
+ *   annual from first possible supplementary
+ * - we wanted every charge version to have a single entry in the results and display
+ *   details for both annual and supplementary in the same entry
+ */
+
+-- [5] Bring the subquery data which returns details of the first annual and supplementary
+-- bill run together with the charge version. We can then display a single line for each
+-- charge version with the matched (nor not) bill run details.
+SELECT
+  r.display_name,
+  cv.licence_ref,
+  cv.charge_version_id,
+  (cv.scheme) AS cv_scheme,
+  (cv.status) AS cv_status,
+  (cv."source") AS cv_source,
```

Comment:
Question: Why is source the only one in a string?

Reply:
The tool I was using ([DBeaver](https://dbeaver.io/)) to write the query did this. My understanding is it does this when you've inadvertently used a protected word for the name of your schema, table or field.